### PR TITLE
Igal/mgmt 1089

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: src/installer/installer.go [hunk 1]
```diff
@@ -34,6 +38,7 @@ const (
 	WaitForControlPlane  = "Waiting for control plane"
 	WritingImageToDisk   = "Writing image to disk"
 	Reboot               = "Rebooting"
+	Joined               = "Joined"
```
Comment (eranco74) [inline src/installer/installer.go:41]: Make sure Jiri knows about it.

File: src/installer/installer.go [hunk 2]
```diff
@@ -250,3 +255,43 @@ func (i *installer) waitForBootkube(ctx context.Context) {
...
+				nodeNameAndCondition[node.Name] = node.Status.Conditions
+				for _, cond := range node.Status.Conditions {
+					if cond.Type == v1.NodeReady && cond.Status == v1.ConditionTrue {
+						readyMasterNodes++
+						if err := i.ic.UpdateHostStatus(Joined, node.Status.NodeInfo.SystemUUID); err != nil {
+							i.log.Errorf("Failed to update node installation status, %s", err)
+						}
+					}
+				}
+			}
```
Comment (eranco74) [inline src/installer/installer.go:283]: Extract to another function

File: src/installer/installer.go [hunk 3]
```diff
@@ -250,3 +255,43 @@ func (i *installer) waitForBootkube(ctx context.Context) {
...
+			i.log.Warnf("Still waiting for master nodes: %v", err)
+		} else {
+			nodeNameAndCondition := map[string][]v1.NodeCondition{}
+			readyMasterNodes := 0
+			for _, node := range nodes.Items {
+				nodeNameAndCondition[node.Name] = node.Status.Conditions
+				for _, cond := range node.Status.Conditions {
+					if cond.Type == v1.NodeReady && cond.Status == v1.ConditionTrue {
+						readyMasterNodes++
+						if err := i.ic.UpdateHostStatus(Joined, node.Status.NodeInfo.SystemUUID); err != nil {
```
Comment (eranco74) [inline src/installer/installer.go:278]: You will send multiple updateHostStatus with Joined.
You should send only one.

File: src/installer/installer.go [hunk 4]
```diff
@@ -250,3 +255,49 @@ func (i *installer) waitForBootkube(ctx context.Context) {
...
+	return nil
+}
+
+func (i *installer) updateReadyMasters(nodes *v1.NodeList, readyMasters []string) []string {
+	nodeNameAndCondition := map[string][]v1.NodeCondition{}
+	for _, node := range nodes.Items {
+		nodeNameAndCondition[node.Name] = node.Status.Conditions
+		for _, cond := range node.Status.Conditions {
+			if cond.Type == v1.NodeReady && cond.Status == v1.ConditionTrue &&
+				!utils.IsElementInStringList(readyMasters, node.Status.NodeInfo.SystemUUID) {
```
Comment (eranco74) [inline src/installer/installer.go:292]: Use https://github.com/thoas/go-funk
Example: https://github.com/filanov/bm-inventory/blob/e6891315968ecc10ce437eec90533790822b8b19/internal/cluster/cluster.go#L173

File: src/installer/installer.go [hunk 5]
```diff
@@ -250,3 +255,49 @@ func (i *installer) waitForBootkube(ctx context.Context) {
...
+		}
+	}, 5*time.Second, apiContext.Done())
+	err := apiContext.Err()
+	if err != nil && err != context.Canceled {
+		return errors.Wrap(err, "Waiting for master nodes")
+	}
+	return nil
+}
+
+func (i *installer) updateReadyMasters(nodes *v1.NodeList, readyMasters []string) []string {
```
Comment (eranco74) [inline src/installer/installer.go:286]: If you will pass a pointer to readyMasters you won't need to return it...

File: src/installer/installer.go [hunk 6]
```diff
@@ -250,3 +255,49 @@ func (i *installer) waitForBootkube(ctx context.Context) {
...
+	nodeNameAndCondition := map[string][]v1.NodeCondition{}
+	for _, node := range nodes.Items {
+		nodeNameAndCondition[node.Name] = node.Status.Conditions
+		for _, cond := range node.Status.Conditions {
+			if cond.Type == v1.NodeReady && cond.Status == v1.ConditionTrue &&
+				!utils.IsElementInStringList(readyMasters, node.Status.NodeInfo.SystemUUID) {
+				readyMasters = append(readyMasters, node.Status.NodeInfo.SystemUUID)
```
Comment (eranco74) [inline src/installer/installer.go:293]: Add log: "Found a new ready master nodes %s ..."

File: src/installer/installer.go [hunk 7]
```diff
@@ -250,3 +255,49 @@ func (i *installer) waitForBootkube(ctx context.Context) {
...
+				if err := i.ic.UpdateHostStatus(Joined, node.Status.NodeInfo.SystemUUID); err != nil {
+					i.log.Errorf("Failed to update node installation status, %s", err)
+				}
+			}
+		}
+	}
+	i.log.Infof("Found %d master nodes: %+v", len(nodes.Items), nodeNameAndCondition)
+	i.log.Infof("Found %d ready master nodes", len(readyMasters))
```
Comment (eranco74) [inline src/installer/installer.go:301]: Consider moving to line 272

File: src/installer/installer.go (hunk 4 above)
Comment (tsorya) [reply src/installer/installer.go:292]: cool , thanks

**Pull request merged**